lib/util/fs/util.ts

```diff
@@ -0,0 +1,33 @@
+import upath from 'upath';
+import { GlobalConfig } from '../../config/global';
+import { FILE_ACCESS_VIOLATION_ERROR } from '../../constants/error-messages';
+import { logger } from '../../logger';
+
+function assertBaseDir(path: string, baseDir: string): void {
+  if (!path.startsWith(upath.resolve(baseDir))) {
+    logger.warn(
+      { path, baseDir },
+      'Preventing access to file outside the base directory'
+    );
+    throw new Error(FILE_ACCESS_VIOLATION_ERROR);
+  }
+}
+
+function ensurePath(path: string, key: 'localDir' | 'cacheDir'): string {
+  const baseDir = upath.resolve(GlobalConfig.get(key)!);
+  let fullPath = path;
+  if (fullPath.startsWith(baseDir)) {
+    fullPath = fullPath.replace(baseDir, '');
+  }
```

Comment:
Yes, because we're using `privateCacheDir` and pass already resolved paths. At the same time, we also treat absolute paths as relative to `localDir`.

Seems like "private" cache isn't related to any caching, mostly being used as the way to have temporary folders/files.